src/components/Personalization/createFetchDataHandler.js

```diff
@@ -36,7 +30,6 @@ export default ({
     );
     onRequestFailure(() => {
```

Comment:
you can move this to `createComponent` in the `if (personalizationDetails.shouldFetchData())` block and not pass the `onRequestFailure` here at all.

Reply:
Good idea. I'll make that change.